File: 0904-fruit-into-baskets/0904-fruit-into-baskets.py
class Solution:
    def totalFruit(self, fruits: List[int]) -> int:
        my_fruits = defaultdict(int)
        l = 0
        for fr in fruits:
            my_fruits[fr] += 1
            if len(my_fruits) > 2:
                if my_fruits[fruits[l]] == 1:
                    del my_fruits[fruits[l]]
                else:
                    my_fruits[fruits[l]] -= 1
                l += 1
        return len(fruits) - l

File: 0904-fruit-into-baskets/0904-fruit-into-baskets.cpp
class Solution {
public:
    int totalFruit(vector<int>& fruits) {
        unordered_map<int, int> my_fruits;
        int l = 0;
        for(auto fr: fruits){
            my_fruits[fr] += 1;
            if (my_fruits.size() > 2){
                if(my_fruits[fruits[l]] == 1){
                    my_fruits.erase(fruits[l]);
                }
                else{
                    --my_fruits[fruits[l]];
                }
                ++l;
            }
        }
        return fruits.size() - l;
    }
};